servers/rendering/renderer_rd/forward_clustered/render_forward_clustered.cpp

```diff
@@ -1884,6 +1884,13 @@ void RenderForwardClustered::_render_scene(RenderDataRD *p_render_data, const Co
 	bool debug_sdfgi_probes = get_debug_draw_mode() == RS::VIEWPORT_DEBUG_DRAW_SDFGI_PROBES;
 	bool depth_pre_pass = bool(GLOBAL_GET("rendering/driver/depth_prepass/enable")) && depth_framebuffer.is_valid();
 
+	uint32_t spec_constant_base_flags = 0;
+	{
+		if (!is_environment(p_render_data->environment) || (p_render_data->environment.is_valid() && environment_get_fog_mode(p_render_data->environment) == RS::EnvironmentFogMode::ENV_FOG_MODE_DEPTH)) {
+			spec_constant_base_flags |= 1 << SPEC_CONSTANT_USE_DEPTH_FOG;
+		}
+	}
```

Comment:
Same as above, we just need to check if the environment is valid:
```suggestion
		if (p_render_data->environment.is_valid() && environment_get_fog_mode(p_render_data->environment) == RS::EnvironmentFogMode::ENV_FOG_MODE_DEPTH) {
			spec_constant_base_flags |= 1 << SPEC_CONSTANT_USE_DEPTH_FOG;
		}
	}
```

Note, here we do need the `SPEC_CONSTANT_USE_DEPTH_FOG` since it isn't provided by the shader directly.